### ISSUE TITLE
Data on 2 devices is partially gone (or inivisible?)

### ISSUE THREAD

**Issue by emninewon** [K1rakishou/Kuroba-Experimental#998]
Hello
I'm using KurobaEX on my phone (XIAOMI REDMI NOTE 10 5G) and on Windows 10 with Bluestacks 5. After the recent Beta update both apps partially reset themselves. Most of the settings are okay but the most important part that got removed (or at least it looks like there is nothing there) is my archived data and set up sites. But the data is still within the app (5.77GB taken on phone, for example).
Is there a way to recover this? Or is it a known bug of the latest update and will be fixed soon?
I'm raising awareness to this if it's not happening for other people.
![Screenshot_2024-01-26-20-25-11-531_com github k1rakishou chan](https://github.com/K1rakishou/Kuroba-Experimental/assets/8757760/4e9b431d-8d89-4f4e-87b5-24953829f9d8)
![Screenshot_2024-01-26-20-25-54-868_com miui securitycenter](https://github.com/K1rakishou/Kuroba-Experimental/assets/8757760/c66c5856-d962-4ae8-b636-4e0d283a517b)
![Screenshot_2024-01-26-20-25-06-280_com github k1rakishou chan](https://github.com/K1rakishou/Kuroba-Experimental/assets/8757760/bca5650a-722c-4720-b6f4-c331aa9b94d6)

**Comment by K1rakishou** [comment]
Can you send me your backup? Settings -> Import/Export -> Export backup.

**Comment by emninewon** [comment]
> Can you send me your backup? Settings -> Import/Export -> Export backup.

Email with the backup sent to you.

**Comment by K1rakishou** [comment]
Unfortunately the database is empty. I don't know what happened to it.

**Comment by emninewon** [comment]
So there is no solution? The archive data seems to still be there. There is no way to recover it?
It's not a major loss, but you should look into the cause and prevent it from happening again, both for me and anyone else.

**Comment by K1rakishou** [comment]
No, there is no solution as of now. The downloaded threads are there on the disk but the metadata that is stored in the database that is used for accessing those threads is gone. I'm not sure if it's possible to recreate it based on the data on the disk. The only thing you can do is manually copy your downloaded data from the emulator using `adb`. In order to look into this I would need logs, at least, but they are probably long gone since I don't persist them anywhere. 
There were no changes related to the database in the latest version(s), so I'm really not sure what could cause this. 
By the way, did you update from v1.3.30 to v1.3.31 or was it an earlier version?

**Comment by emninewon** [comment]
> No, there is no solution as of now. The downloaded threads are there on the disk but the metadata that is stored in the database that is used for accessing those threads is gone. I'm not sure if it's possible to recreate it based on the data on the disk. The only thing you can do is manually copy your downloaded data from the emulator using adb.

So there is no solution for the app on the phone?

> In order to look into this I would need logs, at least, but they are probably long gone since I don't persist them anywhere.

It would be best to reinstate keeping them persistent for a long time then, especially if they don't take much space in the memory. (could be compressed anyway)

> By the way, did you update from v1.3.30 to v1.3.31 or was it an earlier version?

I'm not sure, but I update the app constantly, so it probably got upgraded from the previous version and not an older one, if that's what you're asking.
edit: I've got copies of both v1.3.30 to v1.3.31 autosaved in memory so that is most definitely correct

**Comment by K1rakishou** [comment]
Yeah, I should probably start persisting logs in the database so that there is a way to figure out what happened when issues like this one appear.

**Comment by emninewon** [comment]
Okay, I reinstalled the apps to start fresh and will let you know if it happens again.

**Comment by emninewon** [comment]
Hey, I deleted my original final reply because something weird has happened. I wrote that I already reset and reinstalled both apps but I actually reset only the phone app. Before resetting the app on the emulator (Bluestacks) I decided to backup the settings _just in case_, not thinking I will ever be able to recover it. It was 35GB of data. But after exporting them (together with the archived media), the archive came back. And I see all my stuff that I archived.
So that is a solution that seems to work.
![2024-01-31_01-59-56_HD-Player](https://github.com/K1rakishou/Kuroba-Experimental/assets/8757760/f7667ff7-5899-4a1a-98f0-dc9bad325b05)

**Comment by K1rakishou** [comment]
The hell?

**Comment by emninewon** [comment]
> The hell?

Yeeeep. I lost the archive on my phone but it was small enough (maybe 30 threads from probably a week ago).
But thankfully I decided to export the settings on the emulator and it fixed itself. (I just exported and did nothing else, after it finished all fixed itself)
Edit: oh, and I should mention that I exported the settings before but JUST THE SETTINGS, without the media/archive, and it didn't really do anything (I sent you the backup). This worked only after exporting everything, including the media.

**Comment by K1rakishou** [comment]
Yeah, this feels like "the moon was in a wrong phase" type of a bug.

**Comment by K1rakishou** [comment]
Ok, with your latest backup I have made it so it won't crash now at the very least upon app start. There is some weird inconsistency in the database and because of that it can't auto-delete some old threads and crashes. Will try to figure it out later. 

For now you can wait till this job https://github.com/K1rakishou/Kuroba-Experimental/actions/runs/7766964966/job/21183333338 finishes and then try out the freshly built beta version to see if it works (you can import your backup). 

There release will appear here - https://github.com/K1rakishou/Kuroba-Experimental-beta/releases

**Comment by emninewon** [comment]
Unfortunately this doesn't apply to me since I'm using the Stable version
and not the Beta.
And the crash screen only allows me to export the settings into the backup,
and not the whole (~45.5 GB) archive. So I'm gonna wait till you update the
Stable version for now, and after exporting the whole archive into the
phone memory I might consider switching to the Beta version.

**Comment by emninewon** [comment]
Hey, just wanna let you know that the problem is back. I've lost all the app data 2 times in the past week. I'm gonna finally move to the Beta version.
